contrib/federations_tutorial/authproxy.py

```diff
@@ -0,0 +1,190 @@
+"""
```

Comment:
This seems duplicated with https://github.com/ElementsProject/elements/blob/elements-0.14.1/qa/rpc-tests/test_framework/authproxy.py

Reply:
Yep just added it here for ease. I could hardcode the path to test_framework.

Reply:
My instinct is to always reuse when possible by default, but perhaps it makes sense to duplicate code in this case for simplicity, ping @instagibbs

Reply:
I kind of like the idea of someone just porting the directory and it being usable. I'd say keep it.

Reply:
I've moved this to a separate test_framework dir to make it more obvious. Following your suggestion though to do multi sig (as in #373 ) I had to include half of the `qa/rpc-tests/test_framework` files due to dependencies, thus reusing multiple files not just one. Still okay to do it this way?

Reply:
No, if @instagibbs thinks the duplication makes sense here, I don't have a strong opinion, don't bother.